test/unit/connection/PipelinedWriterTest.js

```diff
@@ -199,16 +201,17 @@ describe('PipelinedWriterTest', function () {
         Promise.all([resolver1.promise, resolver2.promise]).then(() => done());
     });
 
-    it('rejects single promise on write failure', function(done) {
+    it('rejects single promise on write failure', async function() {
         const err = new Error();
         setUpWriteFailure(err);
 
         const resolver = deferredPromise();
         writer.write(createMessageFromString('test'), resolver);
-        resolver.promise.catch((err) => {
-            expect(err).to.be.equal(err);
```

Comment:
This was silly, basically making 1 === 1 assertion. It does not work if I rename one of the variables because we wrap the err in IOError. (in a defected way we pass error as message, fixed that as well.)